README.md

```diff
@@ -1,30 +1,201 @@
+created, as soon as one is needed.
+
+To trigger a dispatcher's processing step, we need to call `poll()` - either
+directly or through a wrapper like `runForever()` or `waitFor()`. This step
+handles any file descriptors, timers and callbacks that are ready to be
+processed.
+
+`Future` objects encapsulate the result of an async procedure, upon successful
+completion, and a list of callbacks to be scheduled after completion or
+cancellation. (These explicit callbacks are rarely used outside Chronos, being
+replaced with the implicit ones generated by async procedure execution and
+`await` chaining.)
+
+Async procedures (those using the `{.async.}` pragma) return `Future` objects.
+
+Inside an async procedure, you can `await` the future returned by another async
+procedure. At this point, control will be handled to the event loop until that
+future is completed.
+
+### Dispatcher
+
+You can run the "dispatcher" event loop forever, with `runForever()` which is defined as:
+
+```nim
+proc runForever*() =
+  while true:
+    poll()
+```
+
+You can also run it until a certain future is completed, with `waitFor()` which
+will also call `Future.read()` on it:
+
+```nim
+proc p(): Future[int] {.async.} =
+  await sleepAsync(100.milliseconds)
+  return 1
+
+echo waitFor p() # prints "1"
+```
+
+`waitFor()` is defined like this:
+
+```nim
+proc waitFor*[T](fut: Future[T]): T =
+  while not(fut.finished()):
+    poll()
+  return fut.read()
+```
+
+### Async procedures and methods
+
+The `{.async.}` pragma will transform a procedure (or a method) returning a
+specialised `Future` type into a closure iterator. If there is no return type
+specified, a `Future[void]` is returned.
+
+```nim
+proc p() {.async.} =
+  await sleepAsync(100.milliseconds)
+
+echo p().type # prints "Future[system.void]"
+```
+
+Whenever `await` is encountered inside an async procedure, control is passed
+back to the dispatcher for as many steps as it's necessary for the awaited
+future to complete (or be cancelled). `await` calls the equivalent of
+`Future.read()` on the completed future and returns the encapsulated value.
+
+```nim
+proc p1() {.async.} =
+  await sleepAsync(1.seconds)
+
+proc p2() {.async.} =
+  await sleepAsync(1.seconds)
+
+proc p3() {.async.} =
+  let
+    fut1 = p1()
+    fut2 = p2()
+  # Just by executing the async procs, both resulting futures entered the
+  # dispatcher's queue and their "clocks" started ticking.
+  await fut1
+  await fut2
+  # Only one second passed while awaiting them both, not two.
+
+waitFor p3()
+```
+
+Not as intuitive, but the same thing happens if we create those futures on the
+same line as `await`, due to the Nim compiler's use of hidden temporary
+variables:
+
+```nim
+proc p4() {.async.} =
+  await p1()
+  await p2()
+  # Also takes a single second for both futures sleeping concurrently.
+
+waitFor p4()
+```
+
+Don't let `await`'s behaviour of giving back control to the dispatcher surprise
```

Comment:
This statement can be easily transformed to more simple one:
> Do not use `await` if you want to be safe.

But if you not using `await` you will be unable to perform concurrent actions.

Reply:
So you make a choice: no changes to the global state or concurrency.

Reply:
> Do not use `await` if you want to be safe.

Too generic to be of any use to newcomers.